### PR TITLE
integrated Psalm by following the instruction given in the issue desc…

### LINKED ISSUE #4: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `psalm.xml` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require --dev vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by paresh27**
…ription

|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by boesing** [comment]
Hey @paresh27, thanks for your first contribution! 🎉 

The builds are failing due to [.travis.yml#L49](https://github.com/laminas-api-tools/api-tools-http-cache/blob/a88c9baf83dabb10995e947f63ea8ded55c14b53/.travis.yml#L49)

You might consider removing this from that line and only add it to the `nightly` matrix entry instead.

**Comment by paresh27** [comment]
@boesing , Shall I make that change in .travi.yml and again generate a PR?

**Comment by boesing** [comment]
@paresh27 No, until all checks are green, we cannot merge this PR. You have to fix `.travis.yml` in this PR.

**Comment by paresh27** [comment]
@boesing , Is it fine now?

**Comment by boesing** [comment]
@paresh27 Sorry for this. I was somehow inaccurate yesterday.

So lets do this step-by-step:
- Remove your latest commit as it breaks `travis.yml` format
- Remove only the `--ignore-platform-reqs` argument from that `travis_retry composer install $COMPOSER_ARGS --ignore-platform-reqs` line
- Let travis build
- If anything is green, awesome. If travis does not build, you probably brick the `.travis.yml` again 😅 . If travis has errors, we have to find out what these errors are and how to fix 'em

File: psalm.xml
Comment (boesing) [inline psalm.xml]: Oh and please rename `.psalm.xml.dist` to `psalm.xml` after removing this file. I have to update the step-by-step guide from the issue as I failed there... again 🙈

**Review by boesing — CHANGES_REQUESTED** [review]

File: psalm.xml
Comment (boesing) [reply psalm.xml]: You mighth ave to re-run `vendor/bin/psalm --set-baseline=psalm-baseline.xml` after the renaming

Comment (paresh27) [reply psalm.xml]: @boesing , I did as per your suggestion. Please let me know if its good now?

**Comment by boesing** [comment]
@paresh27 Sorry that this took so long. I was somehow confused and mixed up stuff.
This PR was about adding psalm to this repository. It had nothing to do with PHP 8.0 - that was my fault and sorry for my misleading reviews.

I've fixed your branch and created #7 to provide PHP 8.0 support to this package (which also drops those versions which could not be tested by travis in this PR due to the old PHP versions).

Thanks for your contribution!

**Comment by weierophinney** [comment]
I've fully rebased this off of current 1.5.x, which contains GHA updates and PHP 8 support. In doing so, it essentially squashes all commits, as rebasing the very first commit resulted in updates that nullified most following commits.

**Pull request merged**
